apps/studio/hooks/misc/useSelectedOrganization.ts

```diff
@@ -4,6 +4,30 @@ import { useMemo } from 'react'
 
 import { LOCAL_STORAGE_KEYS } from 'lib/constants'
 import { useProjectByRef } from './useSelectedProject'
+import { useOrganizationQuery } from 'data/organizations/organization-query'
+
+// [Joshen] Scaffolding this first - will need to double check if this can replace useSelectedOrganization
+export function useSelectedOrganizationV2({ enabled = true } = {}) {
+  const isLoggedIn = useIsLoggedIn()
+  const { ref, slug } = useParams()
+
+  const selectedProject = useProjectByRef(ref)
+  const localStorageSlug = useMemo(() => {
+    return typeof window !== 'undefined'
+      ? localStorage.getItem(LOCAL_STORAGE_KEYS.RECENTLY_VISITED_ORGANIZATION)
+      : null
+  }, [])
+
+  const orgSlug = slug ?? selectedProject?.organization_slug ?? localStorageSlug
+  const { data } = useOrganizationQuery(
+    { slug: orgSlug as string },
+    { enabled: enabled && isLoggedIn && typeof orgSlug === 'string' }
+  )
+
+  return useMemo(() => {
+    return data
+  }, [data, slug, selectedProject, localStorageSlug])
+}
```

Comment:
Is this used in this PR?

Reply:
not yet - but i think i'll look into seeing whether we can use this in a subsequent PR, will keep this PR strictly just for oriole DB changes to make things easier to review and test